internal/operatormanifests/servicemonitor.go

```diff
@@ -0,0 +1,52 @@
+package operatormanifests
+
+import (
+	"fmt"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1alpha1 "github.com/grafana/tempo-operator/apis/config/v1alpha1"
+)
+
+func serviceMonitor(featureGates configv1alpha1.FeatureGates, namespace string) *monitoringv1.ServiceMonitor {
```

Comment:
Shall we move this to a separate prometheus package (the rules as well)?

Reply:
Or I would move the `operatormanifests` under manifest package as `operator`. That makes more sense to me

Reply:
done